src/theory/quantifiers/trigger.h

```diff
@@ -33,102 +32,180 @@ namespace inst {
+  *   (2) the set trNodes is minimal, i.e. removing one term from trNodes always violates (1).
+  */
+  static bool mkTriggerTerms( Node q, std::vector< Node >& nodes, unsigned n_vars, std::vector< Node >& trNodes );
+  /** collectPatTerms
+  * This collects all terms that are eligible for triggers for quantified formula q in term n.
+  *   tstrt is the selection strategy
+  *   exclude is a set of terms that *cannot* be selected as triggers 
+  *   tinfo stores the result of the collection, mapping terms to information they are associated with
+  *   filterInst is a flag that when true, we discard terms that have instances in the vector we are returning.
+  *     e.g. we do not return f( x ) if we are also returning f( f( x ) ).
+  *     TODO: revisit this (issue #1211)
+  */     
   static void collectPatTerms( Node q, Node n, std::vector< Node >& patTerms, quantifiers::TriggerSelMode tstrt,
                                std::vector< Node >& exclude, std::map< Node, TriggerTermInfo >& tinfo,
                                bool filterInst = false );
-  /** is usable trigger */
+  /** is n a usable trigger in quantified formula q? 
+  * A usable trigger is one that is matchable and contains free variables only from q.
+  */
   static bool isUsableTrigger( Node n, Node q );
+  /** return the associated node of n that is a usable trigger in quantified formula q */
   static Node getIsUsableTrigger( Node n, Node q );
+  /** is n a usable atomic trigger? */
   static bool isUsableAtomicTrigger( Node n, Node q );
+  /** is n an atomic trigger? */
   static bool isAtomicTrigger( Node n );
+  /** is k an atomic trigger kind?
+  * An atomic trigger kind is one for which term indexing/matching is possible. 
+  */
   static bool isAtomicTriggerKind( Kind k );
+  /** is n a relational trigger, e.g. like x >= a ? */
   static bool isRelationalTrigger( Node n );
+  /** is k a relational trigger kind? */
   static bool isRelationalTriggerKind( Kind k );
+  /** is k a kind for which counterexample-guided instantiation is possible? */
   static bool isCbqiKind( Kind k );
+  /** is n a simple trigger (see inst_match_generator.h)? */
   static bool isSimpleTrigger( Node n );
+  /** is n a Boolean term trigger, e.g. ite( x, BV1, BV0 )? */
   static bool isBooleanTermTrigger( Node n );
+  /** is n a pure theory trigger, e.g. head( x )? */ 
   static bool isPureTheoryTrigger( Node n );
+  /** get trigger weight
+  * Returns 0 for triggers that are easy to process and 1 otherwise. 
+  */
   static int getTriggerWeight( Node n );
   static bool isLocalTheoryExt( Node n, std::vector< Node >& vars,
                                 std::vector< Node >& patTerms );
-  /** return data structure for producing matches for this trigger. */
-  static InstMatchGenerator* getInstMatchGenerator( Node q, Node n );
+  /** get the variable associated with an inversion for n, e.g. getInversionVariable( x+1 ) returns x */
```

Comment:
I don't understand what you mean by inversion and the example doesn't help.  What does inversion mean here?